sml/src/sml_transport.c

```diff
@@ -108,12 +116,10 @@ void sml_transport_listen(int fd, void (*sml_transport_receiver)(unsigned char *
 	unsigned char buffer[MC_SML_BUFFER_LEN];
 	size_t bytes;
 
-	while (1) {
+	bytes = sml_transport_read(fd, buffer, MC_SML_BUFFER_LEN);
+	while (bytes > 0) {
```

Comment:
there is no need to have two invocations of `sml_transport_read()`.

you could, for example:
```
while ( bytes = sml_transport_read(fd, buffer, MC_SML_BUFFER_LEN) != 0) {
```